patching/EAP7-1063_EAP_Patching_With_Galleon.adoc

```diff
@@ -0,0 +1,124 @@
+
+* not necessary to store content that has been patched or replaced with the version update in a hidden directory in the distribution itself (as the current patching mechanism does), leaving the distribution directory looking like a freshly installed one. (there is still .galleon directory containing the installation provisioning configuration for future reference and a few other text files but not actual heavy JAR content or something of that kind).
+
+While a user still has the possibility to download the feature-pack as a ZIP file and install it locally using Galleon (a la current patching) the repository still remains a general requirement. The reason for that is Galleon does not keep installed feature-packs in the distribution itself but when it's time to re-provision the distribution (and that is happening on every provisioning operation including install feature-pack/patch, uninstall feature-pack/patch, provision specific state, apply update) it assumes the neccessary feature-packs are available in the repository.
+
+Here is a use-case which demonstrates the limitation. User installs EAP (that, btw, currently means Galleon will install three feature-packs: core, servlet and full). Then the user downloads a patch (which is also a kind of feature-pack) and installs it directly, not going through the repository. First of all, what has been installed before the patch (core, servlet and full feature-packs) has to be available in the repository at patch application time. If those feature-packs are not available, patch application will fail. (It is still possible to directly provide core, servlet, full and the patch feature-packs to Galleon for provisioning, although that seems like an impractical way to promote the use of the tool).
+When Galleon installs a feature-pack that is provided directly by the user, by default, besides provisioning that feature-pack, it also installs the directly provided feature-pack into the repository, so it is available for future use (in our case it's a local Maven repository). The user may instruct not to install the feature-pack into the repository but it is strongly discouraged to do that.
+
+==== Patch feature-pack
+
+Patch file itself is basically a feature-pack. The only difference comparing to regular feature-packs is it contains an element in its XML description indicating that it is a patch that applies to a specific version of another feature-pack. A patch will be accepted only if the installation includes the feature-pack the patch applies to. It is possible to install multiple different patches for the same feature-pack in the installation. When the feature-pack that was patched is installed or a different (lower or higher) version of it is installed, all the installed patches for that feature-pack will automatically be removed from the installation.
```

Comment:
Do not confuse one-off patches (unsupported) by async z-stream delivery.

Reply:
Do I? Z-stream is a micro version update, right? So to me, technically, it falls into the version update category. Am I wrong?